pilot/pkg/model/config.go

```diff
@@ -779,6 +799,30 @@ func (store *istioConfigStore) VirtualServices(gateways []string) []Config {
 		}
 	}
 
+	// Need to parse each rule and convert the shortname to FQDN
```

Comment:
I wonder if this could be pushed to a lower level? We should be able to do this processing as soon as we have the ConfigMeta.

Reply:
Hence the commented out code in ConvertObject. I tried that trick and lots of other things seemed to fail, implying that parts of code still relies on ResolveHostname and friends.